### PR TITLE
Mailing feature

### PR DESCRIPTION
**Pull request by JiahuiChen99**
### **New Feature**
Each generated certificate gets sent to the recipient. It uses Yagmail, for now only Gmail senders are supported.
Dependencies updated to the latest version, almost all of them.
> PDF form fields **MUST** have the "Read Only - Sólo Lectura" option checked!

### **Update**
Replaced `PDFJinja` with `pdfrw`.
Fixed a variable comparison bug
```
len_fields2fill = len(fields2fill)
        len_csv_columns = len(next(csv_data))
        if len_fields2fill is not len_csv_columns:
            return "Columns error"
```

### CODE REVIEW

File: django_pdf_form_filler/settings.py [hunk 1]
```diff
@@ -45,15 +45,16 @@ def relative_project_path(*x):
...
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
-    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
+    # 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
     'django.middleware.security.SecurityMiddleware',
+    # 'whitenoise.middleware.WhiteNoiseMiddleware',
```
Comment (cbjuan) [inline django_pdf_form_filler/settings.py]: This may be removed
```suggestion
```

File: django_pdf_form_filler/settings.py [hunk 2]
```diff
@@ -74,7 +75,7 @@ def relative_project_path(*x):
     },
 ]
 
-WSGI_APPLICATION = 'django_pdf_form_filler.wsgi.application'
+# WSGI_APPLICATION = 'application'
```
Comment (cbjuan) [inline django_pdf_form_filler/settings.py]: If not used, it may be removed
```suggestion
```

File: generator/views.py [hunk 3]
```diff
@@ -78,36 +82,56 @@ def save_file(request, file, type):
...
+                                annotation.update(
+                                    pdfrw.PdfDict(V='{}'.format(dict_temp[key]))
+                                )
+                            # Lock the PDF form
+                            # annotation.update(pdfrw.PdfDict(Ff=1))
+            template_pdf.Root.AcroForm.update(pdfrw.PdfDict(NeedAppearances=pdfrw.PdfObject('true')))
+            pdfrw.PdfWriter().write(filled_forms_path + dict_temp['Name'] + '_teem.pdf', template_pdf)
+
+            # Sends the email
+            send_mail(dict_temp['Mail'], "TEEM'21", dict_temp['Name'], filled_forms_path + dict_temp['Name'] + '_teem.pdf')
```
Comment (cbjuan) [inline generator/views.py]: Some of these values are hardcoded (`TEEM'21`, `filled_forms_path + dict_temp['Name'] + '_teem.pdf'`) and may change depending on the subject of the PDF to fill. Can we make them configurable?

File: generator/views.py [hunk 4]
```diff
@@ -78,36 +82,56 @@ def save_file(request, file, type):
...
-            #    return "Filling error"
+
     return filled_forms_path
 
 
 def serve_zip_clean(pdf_file, csv_file, filled_forms_path):
     temp = tempfile.TemporaryFile()
     zf = zipfile.ZipFile(temp, 'w', zipfile.ZIP_DEFLATED)
-    files2zip_list = glob.glob(filled_forms_path + 'filled_form_*.pdf')
+    files2zip_list = glob.glob(filled_forms_path + '*_teem.pdf')
```
Comment (cbjuan) [inline generator/views.py]: Again. Can you make the hardcoded values configurable?

**Review by cbjuan — COMMENTED** [review]
Thank you for the PR! I'm glad you find this repo so useful to spend time updating everything and adding new features.

I added some comments to the PR to help you polish some things. In general terms, I'm OK with adding the new feature and with the cleanup/update you included.

One question. Does this PR work with Python 3? Last time I used the project it worked with Python 2.

Thanks again. Lets keep in touch

**Comment by JiahuiChen99** [comment]
I'll modify the PR according to what you commented in the code review. For now, I hope that it works for TEEM 21 congress.
I'll make those hardcoded values to be configurable 👍 

I had to update most of the libraries and dependencies because when I ran it in my machine, some of those were deprecated and were throwing warnings and issues. Specially, PDFJinja which caused some major problems. I replaced it with pdfwr.

This version works with Python 3.

> Thank you for the PR! I'm glad you find this repo so useful to spend time updating everything and adding new features.
> 
> I added some comments to the PR to help you polish some things. In general terms, I'm OK with adding the new feature and with the cleanup/update you included.
> 
> One question. Does this PR work with Python 3? Last time I used the project it worked with Python 2.
> 
> Thanks again. Lets keep in touch

**Comment by JiahuiChen99** [comment]
Hi @cbjuan !
Sorry for the late modifications, I've been busy finishing my bachelor degree 😅 

I added the suggested modifications and also improved the application. Here you have a rundown of the changes. 

### This PR includes the following features and fixes

- Conference name form property
- Auto-mailing enabling
- Improved form error highlighting
- Mailing feature
- Unused, deprecated dependencies removal

On application start-up, it loads the mailing credentials from os environment variables defined in the **.env** configuration file. The file must be located in the **mailing** folder.

```bash
django_pdf_form_filler
├───...
├───mailing
│   └───.env
├───...
```

Two variables must be set in the .env config file:

- SENDER_EMAIL: Gmail account
- SENDER_PSW: Gmail password

**Review by cbjuan — APPROVED** [review]
Thanks for your PR, @JiahuiChen99. Changes look good to me

**Comment by JiahuiChen99** [comment]
Great, thank you @cbjuan !

**Pull request merged**